win32ss/user/ntuser/sysparams.c

```diff
@@ -346,11 +345,25 @@ SpiUpdatePerUserSystemParameters(VOID)
     gdwLanguageToggleKey = UserGetLanguageToggle();
 }
 
+/* Is Window Snap enabled? */
+static BOOL IntIsWindowSnapEnabled(VOID)
+{
+    WCHAR szValue[2];
+    if (RegReadUserSetting(L"Control Panel\\Desktop", L"WindowArrangementActive",
+                           REG_SZ, szValue, sizeof(szValue)))
+    {
+        szValue[RTL_NUMBER_OF(szValue) - 1] = UNICODE_NULL; /* Avoid buffer overrun */
+        return (_wtoi(szValue) != 0);
+    }
+    return TRUE;
+}
+
 BOOL
 InitSysParams(VOID)
 {
     SpiUpdatePerUserSystemParameters();
     gbSpiInitialized = TRUE;
+    g_bWindowSnapEnabled = IntIsWindowSnapEnabled();
```

Comment:
Considering it reads from the user setting, should it go into `SpiUpdatePerUserSystemParameters` ?

Reply:
Moved.